src/examples/videoUploading.ts

```diff
@@ -0,0 +1,183 @@
+import glslangModule from '../glslang';
+
+export const title = 'Video Texture';
+export const description = 'This example shows how to upload video frame to WebGPU.';
+
+export async function init(canvas: HTMLCanvasElement) {
+  // Set video element
+  const video = document.createElement('video');
+  video.loop = true;
+  video.autoplay = true;
+  video.muted = true;
+  video.src = "assets/video/big-buck-bunny_trailer.webm";
+  await video.play();
+
+  const vertexShaderGLSL = `#version 450
+    layout(location = 0) in vec3 position;
+    layout(location = 1) in vec2 uv;
+
+    layout(location = 0) out vec2 fragUV;
+
+    void main() {
+      gl_Position = vec4(position, 1.0);
+      fragUV = uv;
+    }
+  `;
+
+  const fragmentShaderGLSL = `#version 450
+    layout(set = 0, binding = 0) uniform sampler mySampler;
+    layout(set = 0, binding = 1) uniform texture2D myTexture;
+
+    layout(location = 0) in vec2 fragUV;
+    layout(location = 0) out vec4 outColor;
+
+    void main() {
+        outColor = texture(sampler2D(myTexture, mySampler), fragUV);
+    }
+  `;
+
+  const adapter = await navigator.gpu.requestAdapter();
+  const device = await adapter.requestDevice();
+  const glslang = await glslangModule();
+  const context = canvas.getContext('gpupresent');
+
+  const swapChainFormat = "bgra8unorm";
+
+  var rectVerts = new Float32Array([
+    1.0,  1.0, 0.0, 1.0, 0.0,
+    1.0, -1.0, 0.0, 1.0, 1.0,
+    -1.0, -1.0, 0.0, 0.0, 1.0,
+    1.0,  1.0, 0.0, 1.0, 0.0,
+    -1.0, -1.0, 0.0, 0.0, 1.0,
+    -1.0,  1.0, 0.0, 0.0, 0.0,
+  ]);
+
+  const verticesBuffer = device.createBuffer({
+    size: rectVerts.byteLength,
+    usage: GPUBufferUsage.VERTEX,
+    mappedAtCreation: true,
+  });
+  new Float32Array(verticesBuffer.getMappedRange()).set(rectVerts);
+  verticesBuffer.unmap();
+
+  const bindGroupLayout = device.createBindGroupLayout({
+    entries: [{
+      // Sampler
+      binding: 0,
+      visibility: GPUShaderStage.FRAGMENT,
+      type: "sampler"
+    }, {
+      // Texture view
+      binding: 1,
+      visibility: GPUShaderStage.FRAGMENT,
+      type: "sampled-texture"
+    }]
+  });
+
+  // @ts-ignore:
+  const swapChain = context.configureSwapChain({
+    device,
+    format: swapChainFormat,
+  });
+
+  const pipeline = device.createRenderPipeline({
+    layout: device.createPipelineLayout({ bindGroupLayouts: [bindGroupLayout] }),
+
+    vertexStage: {
+      module: device.createShaderModule({
+        code: glslang.compileGLSL(vertexShaderGLSL, "vertex"),
+
+        // @ts-ignore
+        source: vertexShaderGLSL,
+        transform: source => glslang.compileGLSL(source, "vertex"),
+      }),
+      entryPoint: "main"
+    },
+    fragmentStage: {
+      module: device.createShaderModule({
+        code: glslang.compileGLSL(fragmentShaderGLSL, "fragment"),
+
+        // @ts-ignore
+        source: fragmentShaderGLSL,
+        transform: source => glslang.compileGLSL(source, "fragment"),
```

Comment:
fyi: I'm going to update this so you can do this instead:  no need for ts-ignore as well

```suggestion
        code: fragmentShaderGLSL,
        transform: (glsl) => glslang.compileGLSL(glsl), "fragment"),
```